lib/logproto/logproto-framed-proxied-text-server.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023 One Identity LLC.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "logproto-framed-proxied-text-server.h"
+#include "logproto-proxied-text-server.h"
+#include "logproto-proxied-text-server-private.h"
+#include "logproto-framed-server-private.h"
+
+typedef struct _LogProtoFramedProxiedTextServer
+{
+  LogProtoProxiedTextServer super;
+  LogProtoFramedServer *framed_server;
+} LogProtoFramedProxiedTextServer;
```

Comment:
In my opinion, framed and Text servers should not be mixed as they are orthogonal in terms of functionality.

A Text server is responsible for extracting messages from an EOM-delimited stream, and it uses some clever buffering to do that in an efficient manner.

A Framed server is based on frame/header/octet-counting, and thanks to its simplicity, it does not use such tricky buffering methods.

If I understand correctly, your intention here was "code reuse" rather than mixing the two implementations.

Without going too much into the details of the implementation, I would suggest the following:
- rename this class to `LogProtoProxiedFramedServer`, as Text and Framed just can't go together
- try to inherit from `LogProtoFramedServer` only, and either extract the necessary proxy-extraction functionality from the text server or just reimplement it.

Or anything else that gives a good result in terms of SOLID principles can work too.

If you encounter any difficulties, I think it may worth considering Bazsi's [comment](https://github.com/syslog-ng/syslog-ng/pull/4517#issuecomment-1599239323) and bringing this functionality to the LogTransport level.